int32 publish_count
int32 last_drive_val
int32 last_turn_val 
int32 last_command_count

